### PR TITLE
docs(appmesh): Suggest improvements for App Mesh readme

### PR DESCRIPTION
**Pull request by ghost**
Did a quick review of this README and made some generally minor suggestions.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by Seiya6329** [comment]
Thanks @Jerry-AWS for review the `README`!

**Comment by skinny85** [comment]
@dfezzie would you mind reviewing this PR? Thanks!

**Comment by Seiya6329** [comment]
@Jerry-AWS - Would it be easier for you if I submit separate PR based on your change since there is a merge conflict?

**Comment by Jerry-AWS** [comment]
@Seiya6329 fixed merge conflicts.

File: packages/@aws-cdk/aws-appmesh/README.md [hunk 1]
```diff
@@ -611,19 +608,18 @@ const gateway = mesh.addVirtualGateway('gateway', {
...
 
-A _gateway route_ is attached to a virtual gateway and routes traffic to an existing virtual service.
-If a route matches a request, it can distribute traffic to a target virtual service.
+A _gateway route_ is attached to a virtual gateway and routes matching traffic to an existing virtual service.
 
-For HTTP based gateway routes, the match field can be used to match on 
+For HTTP-based gateway routes, the `match` field can be used to match on 
 path (prefix, exact, or regex), HTTP method, host name, HTTP headers, and query parameters.
-By default, an HTTP based route will match all requests.
+By default, HTTP-based routes match all requests.
```
Comment (Seiya6329) [inline packages/@aws-cdk/aws-appmesh/README.md:622]:
```suggestion
By default, HTTP-based gateway routes match all requests.
```

Comment (Seiya6329) [reply packages/@aws-cdk/aws-appmesh/README.md:622]: Sorry, above is my miss from the earlier PR.

File: packages/@aws-cdk/aws-appmesh/README.md [hunk 2]
```diff
@@ -216,22 +214,22 @@ const virtualService = new appmesh.VirtualService(stack, 'service-1', {
 node.addBackend(appmesh.Backend.virtualService(virtualService));
 ```
 
-The `listeners` property can be left blank and added later with the `node.addListener()` method. The `healthcheck` and `timeout` properties are optional but if specifying a listener, the `port` must be added.
+The `listeners` property can be left blank and added later with the `node.addListener()` method. The `healthcheck` and `timeout` properties are optional, but the `port` must be specified when specifying a listener.
```
Comment (Seiya6329) [inline packages/@aws-cdk/aws-appmesh/README.md:217]:
```suggestion
The `listeners` property can be left blank and added later with the `node.addListener()` method. The `serviceDiscovery ` must be specified when specifying a listener.
```

Actually `port` is also optional even when using `node.addListener()`.

I think `healthcheck` and `timeout` reference can also be removed. Even though they actually are optional properties for `listener`, all properties for `listener` are optional.

Instead, I think useful information to add here is the call out that the `serviceDiscovery` for a `VirtualNode` must be specified when `listeners` is specified. ([reference](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appmesh-virtualnode-virtualnodespec.html))

**Comment by Seiya6329** [comment]
@Jerry-AWS - Thanks for the review. It is looking great and much easier to read. I have minor comments.

File: packages/@aws-cdk/aws-appmesh/README.md (hunk 2 above)
Comment (Seiya6329) [reply packages/@aws-cdk/aws-appmesh/README.md:217]: To mention, this discrepancy should've been addressed earlier.

File: packages/@aws-cdk/aws-appmesh/README.md [hunk 3]
```diff
@@ -392,19 +389,19 @@ const gateway = new appmesh.VirtualGateway(this, 'gateway', {
 
 ## Adding a Route
 
-A `route` is associated with a virtual router, and it's used to match requests for a virtual router and distribute traffic accordingly to its associated virtual nodes.
+A _route_ matches requests with an associated virtual router and distributes traffic to its associated virtual nodes. 
+The route distributes matching requests to one or more target virtual nodes with relative weighting.
 
-If your `route` matches a request, you can distribute traffic to one or more target virtual nodes with relative weighting.
+The `RouteSpec` class lets you define protocol-specific route specifications.
+The `tcp()`, `http()` and `http2()` methods methods create a specification for the named protocols.
```
[the remark this answers is not in the document]
Comment (Jerry-AWS) [reply packages/@aws-cdk/aws-appmesh/README.md]: Will also add backticks around grpc()

**Comment by Jerry-AWS** [comment]
@Seiya6329 Accepted those changes; this should be good to go.

File: packages/@aws-cdk/aws-appmesh/README.md [hunk 4]
```diff
@@ -561,12 +558,12 @@ router.addRoute('route-http', {
...
 
-A _virtual gateway_ allows resources outside your mesh to communicate to resources that are inside your mesh.
+A _virtual gateway_ allows resources outside your mesh to communicate with resources inside your mesh.
 The virtual gateway represents an Envoy proxy running in an Amazon ECS task, in a Kubernetes service, or on an Amazon EC2 instance.
-Unlike a virtual node, which represents an Envoy running with an application, a virtual gateway represents Envoy deployed by itself.
+Unlike a virtual node, which represents Envoy running with an application, a virtual gateway represents Envoy deployed by itself.
 
 A virtual gateway is similar to a virtual node in that it has a listener that accepts traffic for a particular port and protocol (HTTP, HTTP2, GRPC).
-The traffic that the virtual gateway receives, is directed to other services in your mesh,
+Traffic recieved by the virtual gateway is directed to other services in your mesh
```
[the remark this answers is not in the document]
Comment (Seiya6329) [reply packages/@aws-cdk/aws-appmesh/README.md:564]: @Jerry-AWS Looks like you reverted back the change.

File: packages/@aws-cdk/aws-appmesh/README.md (hunk 1 above)
Comment (Seiya6329) [reply packages/@aws-cdk/aws-appmesh/README.md:620]: @Jerry-AWS Same here. Looks like you reverted back the change.

File: packages/@aws-cdk/aws-appmesh/README.md (hunk 2 above)
Comment (Seiya6329) [reply packages/@aws-cdk/aws-appmesh/README.md:217]: @Jerry-AWS Looks like you reverted back the change. Was this interntional?

Comment (Seiya6329) [reply packages/@aws-cdk/aws-appmesh/README.md:217]: @Jerry-AWS Looks like you reverted back the change. Was this intentional? #

**Comment by Seiya6329** [comment]
@Jerry-AWS Thanks for the update! Looks like you have overwritten following changes during the last commit. Were they intentional?
- https://github.com/aws/aws-cdk/pull/15506#discussion_r674241064
- https://github.com/aws/aws-cdk/pull/15506#discussion_r674241275
- https://github.com/aws/aws-cdk/pull/15506#discussion_r674244216

**Comment by Jerry-AWS** [comment]
@Seiya6329 No, unintentional. I'll restore those changes.

**Comment by Jerry-AWS** [comment]
@Seiya6329 The inadvertent reversions have been addressed and it's building now. Should be good to go!

**Comment by Seiya6329** [comment]
@Jerry-AWS Looks great! Thank you very much!

**Comment by Seiya6329** [comment]
@skinny85 I reviewed the change and it looks good. Could you approve it to merge?

**Review by skinny85 — APPROVED** [review]

**Pull request merged**